src/components/shared/passphraseRenderer/index.js

```diff
@@ -0,0 +1,231 @@
+import React from 'react';
+import fillWordsList from 'bitcore-mnemonic/lib/words/english';
+import grid from 'flexboxgrid/dist/flexboxgrid.css';
+import { withTranslation } from 'react-i18next';
+import styles from './passphraseRenderer.css';
+import { PrimaryButton, TertiaryButton } from '../../toolbox/buttons/button';
+
+class PassphraseRenderer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      indexes: [2, 9],
+      fieldSelected: undefined,
+      displayOptions: undefined,
+      chosenWords: {},
+      options: {},
+      isCorrect: false,
+      hasErrors: false,
+      disabledButton: true,
+    };
+
+    this.handleConfirm = this.handleConfirm.bind(this);
+  }
+
+  UNSAFE_componentWillMount() { // eslint-disable-line camelcase
+    const { indexes } = this.state;
+
+    const options = this.assembleWordOptions(this.props.values, indexes);
+    this.setState({
+      ...this.state,
+      options: {
+        [indexes[0]]: options[0],
+        [indexes[1]]: options[1],
+      },
+    });
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.hasErrors && !this.props.hasErrors) {
+      this.setState({
+        fieldSelected: undefined,
+        displayOptions: undefined,
+        chosenWords: {},
```

Comment:
I don't see why this is needed. AFAIK `hasErrors` is in `state`, not in `props`.

Reply:
True. I forgot to remove it after moving the logic from the screen to the component.